### PR TITLE
Add 3.3.0 release note and news and update links

### PR DESCRIPTION
**Pull request by MaxGekk**
This PR aims to update our website. I checked the followings are available as of now.

https://spark.apache.org/docs/3.3.0/
https://dist.apache.org/repos/dist/release/spark/spark-3.3.0/
https://repo1.maven.org/maven2/org/apache/spark/spark-core_2.12/3.3.0/
https://pypi.org/project/pyspark/3.3.0/

### CODE REVIEW

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM.

File: releases/_posts/2022-06-16-spark-release-3-3-0.md [hunk 1]
```diff
@@ -0,0 +1,477 @@
...
+
+
+### Spark SQL and Core
+
+
+#### ANSI mode
+
+
+
+* New explicit cast syntax rules in ANSI mode ([SPARK-33354](https://issues.apache.org/jira/browse/SPARK-33354))
```
Comment (cloud-fan) [inline releases/_posts/2022-06-16-spark-release-3-3-0.md:48]: I checked the JIRA and it was merged to Spark 3.1.

File: releases/_posts/2022-06-16-spark-release-3-3-0.md [hunk 2]
```diff
@@ -0,0 +1,477 @@
...
+* Add a config to allow casting between Datetime and Numeric ([SPARK-37179](https://issues.apache.org/jira/browse/SPARK-37179))
+* Add a config to optionally enforce ANSI reserved keywords ([SPARK-37133](https://issues.apache.org/jira/browse/SPARK-37133))
+* Disallow binary operations between Interval and String literal ([SPARK-36508](https://issues.apache.org/jira/browse/SPARK-36508))
+
+
+#### Feature Enhancements
+
+
+
+* Support ANSI SQL INTERVAL types ([SPARK-27790](https://issues.apache.org/jira/browse/SPARK-27790))
```
Comment (cloud-fan) [inline releases/_posts/2022-06-16-spark-release-3-3-0.md:63]: I don't think this is a 3.3 feature, but we support writing out interval to data sources in 3.3

File: releases/_posts/2022-06-16-spark-release-3-3-0.md [hunk 3]
```diff
@@ -0,0 +1,477 @@
...
+* Error Message Improvements ([SPARK-38781](https://issues.apache.org/jira/browse/SPARK-38781))
+* Hidden File Metadata Support for Spark SQL ([SPARK-37273](https://issues.apache.org/jira/browse/SPARK-37273))
+* Support raw string literal ([SPARK-36371](https://issues.apache.org/jira/browse/SPARK-36371))
+* Helper class for batch Dataset.observe() ([SPARK-34806](https://issues.apache.org/jira/browse/SPARK-34806))
+* Support specify initial partition number for rebalance ([SPARK-38410](https://issues.apache.org/jira/browse/SPARK-38410))
+* Support cascade mode for `dropNamespace` API ([SPARK-37929](https://issues.apache.org/jira/browse/SPARK-37929))
+* Allow store assignment and implicit cast among datetime types ([SPARK-37707](https://issues.apache.org/jira/browse/SPARK-37707))
+* Collect, first and last should be deterministic aggregate functions ([SPARK-32940](https://issues.apache.org/jira/browse/SPARK-32940))
+* Add ExpressionBuilder for functions with complex overloads ([SPARK-37164](https://issues.apache.org/jira/browse/SPARK-37164))
```
Comment (cloud-fan) [inline releases/_posts/2022-06-16-spark-release-3-3-0.md]: This is a refactor. Do we need to mention it in release notes? It's not user facing.

File: releases/_posts/2022-06-16-spark-release-3-3-0.md [hunk 4]
```diff
@@ -0,0 +1,477 @@
...
+* Add array support to union by name ([SPARK-36546](https://issues.apache.org/jira/browse/SPARK-36546))
+* Add df.withMetadata: a syntax sugar to update the metadata of a dataframe ([SPARK-36642](https://issues.apache.org/jira/browse/SPARK-36642))
+* Support raw string literal ([SPARK-36371](https://issues.apache.org/jira/browse/SPARK-36371))
```
Comment (cloud-fan) [inline releases/_posts/2022-06-16-spark-release-3-3-0.md]: this duplicated line 66

File: releases/_posts/2022-06-16-spark-release-3-3-0.md [hunk 5]
```diff
@@ -0,0 +1,477 @@
...
+* Add AS OF syntax support ([SPARK-37219](https://issues.apache.org/jira/browse/SPARK-37219))
+* Add REPEATABLE in TABLESAMPLE to specify seed ([SPARK-37165](https://issues.apache.org/jira/browse/SPARK-37165))
+* Add ansi syntax `set catalog xxx` to change the current catalog ([SPARK-36841](https://issues.apache.org/jira/browse/SPARK-36841))
+* Support ILIKE (ALL | ANY | SOME) - case insensitive LIKE ([SPARK-36674](https://issues.apache.org/jira/browse/SPARK-36674), [SPARK-36736](https://issues.apache.org/jira/browse/SPARK-36736), [SPARK-36778](https://issues.apache.org/jira/browse/SPARK-36778))
+* Support query stage show runtime statistics in formatted explain mode ([SPARK-38322](https://issues.apache.org/jira/browse/SPARK-38322))
+* Add spill size metrics for sort merge join ([SPARK-37726](https://issues.apache.org/jira/browse/SPARK-37726))
+* Update the SQL syntax of SHOW FUNCTIONS ([SPARK-37777](https://issues.apache.org/jira/browse/SPARK-37777))
+* Implement support for DEFAULT values for columns in tables ([SPARK-38334](https://issues.apache.org/jira/browse/SPARK-38334))
```
Comment (cloud-fan) [inline releases/_posts/2022-06-16-spark-release-3-3-0.md]: I think this is a 3.4 feature, cc @gengliangwang

File: releases/_posts/2022-06-16-spark-release-3-3-0.md [hunk 6]
```diff
@@ -0,0 +1,477 @@
...
+* Whole-stage code generation
+    * Add code-gen for sort aggregate without grouping keys ([SPARK-37564](https://issues.apache.org/jira/browse/SPARK-37564))
+    * Add code-gen for full outer sort merge join ([SPARK-35352](https://issues.apache.org/jira/browse/SPARK-35352))
+    * Add code-gen for full outer shuffled hash join ([SPARK-32567](https://issues.apache.org/jira/browse/SPARK-32567))
+    * Add code-gen for existence sort merge join ([SPARK-37316](https://issues.apache.org/jira/browse/SPARK-37316))
+* Push down (filters)
+    * Push down filters through RebalancePartitions ([SPARK-37828](https://issues.apache.org/jira/browse/SPARK-37828))
+    * Push down boolean column filter ([SPARK-36644](https://issues.apache.org/jira/browse/SPARK-36644))
+    * Push down limit 1 for right side of left semi/anti join if join condition is empty ([SPARK-37917](https://issues.apache.org/jira/browse/SPARK-37917))
+    * Translate more standard aggregate functions for pushdown ([SPARK-37527](https://issues.apache.org/jira/browse/SPARK-37527))
```
Comment (cloud-fan) [inline releases/_posts/2022-06-16-spark-release-3-3-0.md]: We should put this in the `Data Source V2 API` section.

File: releases/_posts/2022-06-16-spark-release-3-3-0.md (hunk 1 above)
Comment (MaxGekk) [reply releases/_posts/2022-06-16-spark-release-3-3-0.md:48]: @cloud-fan Could you leave comments in https://docs.google.com/document/d/1gGySrLGvIK8bajKdGjTI_mDqk0-YPvHmPN64YjoWfOQ/edit?usp=sharing , please.

File: releases/_posts/2022-06-16-spark-release-3-3-0.md [hunk 7]
```diff
@@ -0,0 +1,477 @@
...
+
+
+
+* General
+    * Lenient serialization of datetime from datasource ([SPARK-38437](https://issues.apache.org/jira/browse/SPARK-38437))
+    * Treat table location as absolute when the first letter of its path is slash in create/alter table ([SPARK-38236](https://issues.apache.org/jira/browse/SPARK-38236))
+    * Remove leading zeros from empty static number type partition ([SPARK-35561](https://issues.apache.org/jira/browse/SPARK-35561))
+    * Support `ignoreCorruptFiles` and `ignoreMissingFiles` in Data Source options ([SPARK-38767](https://issues.apache.org/jira/browse/SPARK-38767))
+* Parquet
+    * Enable matching schema column names by field ids ([SPARK-38094](https://issues.apache.org/jira/browse/SPARK-38094))
```
Comment (cloud-fan) [inline releases/_posts/2022-06-16-spark-release-3-3-0.md:194]: This is a parquet feature.

File: releases/_posts/2022-06-16-spark-release-3-3-0.md [hunk 8]
```diff
@@ -0,0 +1,477 @@
...
+    * Remove check field name when reading/writing data in parquet ([SPARK-27442](https://issues.apache.org/jira/browse/SPARK-27442))
+    * Support vectorized read boolean values use RLE encoding with Parquet DataPage V2 ([SPARK-37864](https://issues.apache.org/jira/browse/SPARK-37864))
+    * Support Parquet V2 data page encoding (DELTA_BINARY_PACKED) for the vectorized path ([SPARK-36879](https://issues.apache.org/jira/browse/SPARK-36879))
+    * Rebase timestamps in the session time zone saved in Parquet/Avro metadata ([SPARK-37705](https://issues.apache.org/jira/browse/SPARK-37705))
+    * Push down group by partition column for aggregate ([SPARK-36646](https://issues.apache.org/jira/browse/SPARK-36646))
+    * Aggregate (Min/Max/Count) push down for Parquet ([SPARK-36645](https://issues.apache.org/jira/browse/SPARK-36645))
+    * Parquet: enable matching schema columns by field id ([SPARK-38094](https://issues.apache.org/jira/browse/SPARK-38094))
```
Comment (cloud-fan) [inline releases/_posts/2022-06-16-spark-release-3-3-0.md]: This duplicates line 194

File: releases/_posts/2022-06-16-spark-release-3-3-0.md (hunk 1 above)
Comment (MaxGekk) [reply releases/_posts/2022-06-16-spark-release-3-3-0.md:48]: After offline discussion, let me address comments directly in the PR.

File: releases/_posts/2022-06-16-spark-release-3-3-0.md [hunk 9]
```diff
@@ -0,0 +1,477 @@
...
+    * Support aggregate functions of build-in JDBC dialect ([SPARK-37867](https://issues.apache.org/jira/browse/SPARK-37867))
+    * Move compileAggregates from JDBCRDD to JdbcDialect ([SPARK-37286](https://issues.apache.org/jira/browse/SPARK-37286))
+    * Implement dropIndex and listIndexes in JDBC (MySQL dialect) ([SPARK-36914](https://issues.apache.org/jira/browse/SPARK-36914))
+    * Supports list namespaces in JDBC V2 MySQL dialect ([SPARK-38054](https://issues.apache.org/jira/browse/SPARK-38054))
+    * Add factory method getConnection into JDBCDialect ([SPARK-38361](https://issues.apache.org/jira/browse/SPARK-38361))
+    * Jdbc dialect should decide which function could be pushed down ([SPARK-39162](https://issues.apache.org/jira/browse/SPARK-39162))
+    * Propagate correct JDBC properties in JDBC connector provider and add "connectionProvider" option ([SPARK-36163](https://issues.apache.org/jira/browse/SPARK-36163))
+    * Refactor framework so as JDBC dialect could compile filter by self way ([SPARK-38432](https://issues.apache.org/jira/browse/SPARK-38432))
```
Comment (cloud-fan) [inline releases/_posts/2022-06-16-spark-release-3-3-0.md:224]:
```suggestion
    * Refactor framework so as JDBC dialect could compile filter by itself ([SPARK-38432](https://issues.apache.org/jira/browse/SPARK-38432))
```

File: releases/_posts/2022-06-16-spark-release-3-3-0.md [hunk 10]
```diff
@@ -0,0 +1,477 @@
...
+
+
+* New interfaces
+    * Introduce a new DataSource V2 interface HasPartitionKey ([SPARK-37376](https://issues.apache.org/jira/browse/SPARK-37376))
+    * Add interface SupportsPushDownV2Filters ([SPARK-36760](https://issues.apache.org/jira/browse/SPARK-36760))
+    * Support DataSource V2 CreateTempViewUsing ([SPARK-35803](https://issues.apache.org/jira/browse/SPARK-35803))
+    * Add a class to represent general aggregate functions in DS V2 ([SPARK-37789](https://issues.apache.org/jira/browse/SPARK-37789))
+    * A new framework to represent catalyst expressions in DS V2 APIs ([SPARK-37960](https://issues.apache.org/jira/browse/SPARK-37960))
+    * Reactor framework so as JDBC dialect could compile expression by self way ([SPARK-38196](https://issues.apache.org/jira/browse/SPARK-38196))
+    * Refactor framework so as JDBC dialect could compile filter by self way ([SPARK-38432](https://issues.apache.org/jira/browse/SPARK-38432))
```
Comment (cloud-fan) [inline releases/_posts/2022-06-16-spark-release-3-3-0.md]: are these 2 belong to the JDBC section?

File: releases/_posts/2022-06-16-spark-release-3-3-0.md [hunk 11]
```diff
@@ -0,0 +1,477 @@
...
+    * DSV2 InputMetrics are not getting update in corner case ([SPARK-37585](https://issues.apache.org/jira/browse/SPARK-37585))
```
Comment (cloud-fan) [inline releases/_posts/2022-06-16-spark-release-3-3-0.md]: this is a bug fix, not new interface.

File: releases/_posts/2022-06-16-spark-release-3-3-0.md [hunk 12]
```diff
@@ -0,0 +1,477 @@
...
+    * Add APIs for group-based row-level operations ([SPARK-38625](https://issues.apache.org/jira/browse/SPARK-38625))
+* Migrate commands
+    * Migrate SHOW CREATE TABLE to use V2 command by default ([SPARK-37878](https://issues.apache.org/jira/browse/SPARK-37878))
+    * Migrate CREATE NAMESPACE to use V2 command by default ([SPARK-37636](https://issues.apache.org/jira/browse/SPARK-37636))
+    * Migrate DESCRIBE NAMESPACE to use V2 command by default ([SPARK-37150](https://issues.apache.org/jira/browse/SPARK-37150))
+* Indexing
+    * Support drop index for Data Source V2 ([SPARK-37200](https://issues.apache.org/jira/browse/SPARK-37200))
+    * Add Create Index syntax support ([SPARK-36895](https://issues.apache.org/jira/browse/SPARK-36895))
+    * Implement createIndex and IndexExists in DS V2 JDBC (MySQL dialect) ([SPARK-36913](https://issues.apache.org/jira/browse/SPARK-36913))
```
Comment (cloud-fan) [inline releases/_posts/2022-06-16-spark-release-3-3-0.md]: does it belong to the JDBC section?

File: releases/_posts/2022-06-16-spark-release-3-3-0.md [hunk 13]
```diff
@@ -0,0 +1,477 @@
...
+    * DS V2 Index Support: Add supportsIndex interface ([SPARK-36526](https://issues.apache.org/jira/browse/SPARK-36526))
+* Push down ([SPARK-38788](https://issues.apache.org/jira/browse/SPARK-38788))
+    * Add DS V2 filters ([SPARK-36556](https://issues.apache.org/jira/browse/SPARK-36556))
+    * Push down boolean column filter for Data Source V2 ([SPARK-36644](https://issues.apache.org/jira/browse/SPARK-36644))
+    * Support push down top N to JDBC data source V2 ([SPARK-37483](https://issues.apache.org/jira/browse/SPARK-37483))
+    * DS V2 Sample Push Down ([SPARK-37038](https://issues.apache.org/jira/browse/SPARK-37038))
+    * DS V2 LIMIT push down ([SPARK-37020](https://issues.apache.org/jira/browse/SPARK-37020))
+    * DS V2 supports partial aggregate push-down `AVG` ([SPARK-37839](https://issues.apache.org/jira/browse/SPARK-37839))
+    * Support datasource V2 complete aggregate pushdown ([SPARK-37644](https://issues.apache.org/jira/browse/SPARK-37644))
+    * DS V2 supports partial aggregate push-down `AVG` ([SPARK-37839](https://issues.apache.org/jira/browse/SPARK-37839))
```
Comment (cloud-fan) [inline releases/_posts/2022-06-16-spark-release-3-3-0.md]: it duplicates line 268

File: releases/_posts/2022-06-16-spark-release-3-3-0.md [hunk 14]
```diff
@@ -0,0 +1,477 @@
...
+    * Translate more standard aggregate functions for pushdown ([SPARK-37527](https://issues.apache.org/jira/browse/SPARK-37527))
+    * DS V2 aggregate push-down supports project with alias ([SPARK-38533](https://issues.apache.org/jira/browse/SPARK-38533))
+    * DS V2 topN push-down supports project with alias ([SPARK-38644](https://issues.apache.org/jira/browse/SPARK-38644))
+    * DS V2 Top N push-down supports order by expressions ([SPARK-39037](https://issues.apache.org/jira/browse/SPARK-39037))
+    * Datasource V2 supports partial topN push-down ([SPARK-38391](https://issues.apache.org/jira/browse/SPARK-38391))
+    * Support push down Cast to JDBC data source V2 ([SPARK-38633](https://issues.apache.org/jira/browse/SPARK-38633))
+    * If limit could pushed down and Data source only have one partition, DS V2 should not do limit again ([SPARK-38768](https://issues.apache.org/jira/browse/SPARK-38768))
+    * Remove `Limit` from plan if complete push down limit to data source ([SPARK-38768](https://issues.apache.org/jira/browse/SPARK-38768))
```
Comment (cloud-fan) [inline releases/_posts/2022-06-16-spark-release-3-3-0.md:269]: It duplicates the item above.

File: releases/_posts/2022-06-16-spark-release-3-3-0.md [hunk 15]
```diff
@@ -0,0 +1,477 @@
...
+* Fix correctness issue on stream-stream outer join with RocksDB state store provider ([SPARK-38684](https://issues.apache.org/jira/browse/SPARK-38684))
+* Upgrade Kafka to 3.1.0 ([SPARK-36837](https://issues.apache.org/jira/browse/SPARK-36837))
+* Support Trigger.AvailableNow on Kafka data source ([SPARK-36649](https://issues.apache.org/jira/browse/SPARK-36649))
+* Optimize write path on RocksDB state store provider ([SPARK-37224](https://issues.apache.org/jira/browse/SPARK-37224))
+* Introduce a new data source for providing consistent set of rows per microbatch ([SPARK-37062](https://issues.apache.org/jira/browse/SPARK-37062))
+* Use HashClusteredDistribution for stateful operators with respecting backward compatibility ([SPARK-38204](https://issues.apache.org/jira/browse/SPARK-38204))
+* Make foreachBatch streaming query stop gracefully ([SPARK-39218](https://issues.apache.org/jira/browse/SPARK-39218))
+
+
+### PySpark
```
Comment (cloud-fan) [inline releases/_posts/2022-06-16-spark-release-3-3-0.md:348]: @HyukjinKwon  can you review the pyspark section?

File: releases/_posts/2022-06-16-spark-release-3-3-0.md (hunk 1 above)
Comment (MaxGekk) [reply releases/_posts/2022-06-16-spark-release-3-3-0.md:48]: The ticket is an umbrella for others that were merged to 3.3 in particular, for example:
https://issues.apache.org/jira/browse/SPARK-37179
I have taken the umbrella only to not mention all sub-tasks.

File: releases/_posts/2022-06-16-spark-release-3-3-0.md (hunk 2 above)
Comment (MaxGekk) [reply releases/_posts/2022-06-16-spark-release-3-3-0.md:63]: This is an umbrella too in which the last sub-tasks were done in 3.3 only.

File: releases/_posts/2022-06-16-spark-release-3-3-0.md (hunk 3 above)
Comment (MaxGekk) [reply releases/_posts/2022-06-16-spark-release-3-3-0.md]: ok. I'll remove it.

File: releases/_posts/2022-06-16-spark-release-3-3-0.md (hunk 4 above)
Comment (MaxGekk) [reply releases/_posts/2022-06-16-spark-release-3-3-0.md]: Removed the duplicate.

File: releases/_posts/2022-06-16-spark-release-3-3-0.md (hunk 5 above)
Comment (MaxGekk) [reply releases/_posts/2022-06-16-spark-release-3-3-0.md]: Yep, let me remove it.

File: releases/_posts/2022-06-16-spark-release-3-3-0.md (hunk 6 above)
Comment (MaxGekk) [reply releases/_posts/2022-06-16-spark-release-3-3-0.md]: Moved.

File: releases/_posts/2022-06-16-spark-release-3-3-0.md (hunk 8 above)
Comment (MaxGekk) [reply releases/_posts/2022-06-16-spark-release-3-3-0.md]: Removed the duplicate

File: releases/_posts/2022-06-16-spark-release-3-3-0.md (hunk 14 above)
Comment (MaxGekk) [reply releases/_posts/2022-06-16-spark-release-3-3-0.md:269]: Removed the duplicate.

File: releases/_posts/2022-06-16-spark-release-3-3-0.md (hunk 13 above)
Comment (MaxGekk) [reply releases/_posts/2022-06-16-spark-release-3-3-0.md]: Removed the duplicate.

File: releases/_posts/2022-06-16-spark-release-3-3-0.md (hunk 11 above)
Comment (MaxGekk) [reply releases/_posts/2022-06-16-spark-release-3-3-0.md]: Removed

File: releases/_posts/2022-06-16-spark-release-3-3-0.md (hunk 9 above)
Comment (MaxGekk) [reply releases/_posts/2022-06-16-spark-release-3-3-0.md:224]: Changed

File: releases/_posts/2022-06-16-spark-release-3-3-0.md (hunk 10 above)
Comment (MaxGekk) [reply releases/_posts/2022-06-16-spark-release-3-3-0.md]: Moved to the JDBC section

File: releases/_posts/2022-06-16-spark-release-3-3-0.md (hunk 12 above)
Comment (MaxGekk) [reply releases/_posts/2022-06-16-spark-release-3-3-0.md]: Moved to the JDBC section

File: releases/_posts/2022-06-16-spark-release-3-3-0.md (hunk 15 above)
Comment (MaxGekk) [reply releases/_posts/2022-06-16-spark-release-3-3-0.md:348]: Hyukjin reviewed the original doc https://docs.google.com/document/d/1gGySrLGvIK8bajKdGjTI_mDqk0-YPvHmPN64YjoWfOQ/edit#heading=h.v5lk2k7s3ulx as far as I know.

Comment (gatorsmile) [reply releases/_posts/2022-06-16-spark-release-3-3-0.md:348]: I think Hyukjin has reviewed the PySpark section.

**Review by gatorsmile — APPROVED** [review]

**Comment by MaxGekk** [comment]
Merging to asf-site. Thank you, @dongjoon-hyun @cloud-fan @gatorsmile for review.

**Pull request closed without merging**